packages/experiments/src/utilities/BaseState.tsx

```diff
@@ -31,12 +31,19 @@ export class BaseState<TComponentProps, TViewProps, TState> extends React.Compon
     }
   }
 
+  // Unless overwritten by the component, the props stay the same.
+  public transformDerivedProps(newProps: TViewProps): TViewProps {
```

Comment:
Hmm, I think this would be better as part of options along with controlled props.

Also, the typing seems to provide clarity on the name. I wonder if `transformViewProps` is a better name?